### ISSUE TITLE
Cannot exclude root folders while searching

### ISSUE THREAD

**Issue by sandy081** [Microsoft/vscode#50638]
Testing #50498

- In a MR workspace, try to exclude root folder using **/folder1/** and search. Results are still shown in folder1.

This happens irrespective of setting `search.enableSearchProviders" `

![image](https://user-images.githubusercontent.com/10746682/40652451-8f56d79e-6338-11e8-8994-58f40d5ab239.png)

**Comment by roblourens** [comment]
Yeah glob matching starts under the search folder so there's actually no way to do this...

**Comment by sandy081** [comment]
Is it possible to provide a special syntax for excluding/including root folders?

**Comment by roblourens** [comment]
I would support `./name` in 'exclude' too. It works in 'include'

**Comment by pacifier007** [comment]
Similar to this, it would be useful if the root folder could be referenced, not just in this use-case but in multi-root use-cases as well.

See my use-case that would be resolved easily if we could reference root folders: https://github.com/microsoft/vscode/issues/91102#issuecomment-589244585

Also related: 
#48262
#82145

**Comment by OmeGak** [comment]
What happened to @roblourens' idea of [supporting per-folder settings in multi-root workspaces](https://github.com/Microsoft/vscode/issues/48262#issuecomment-383347845)? It would also be helpful here.

**Comment by unional** [comment]
This is a major issue for multi-root workspace. Currently I'm hitting with the watch file limits (which I cannot change the environment) and there isn't a way to do `files.exclude` on the each root.

**Comment by nsgundy** [comment]
Looking for the same thing, would like to exclude a folder on a particular root that then doesn't get excluded on any of the others.

Something like this:
```
"files.exclude": {
  "${workspaceFolder:rootA}/exclude": true
}
```

**Comment by thomaslaurie** [comment]
This was an annoying issue to track down, I thought search was bugged for the longest time. 

Its an issue for me in large projects because I like to have folders like `node_modules` and `dist` added as root folders for easier access. However this prevents them from being excluded in search results, which is obviously undesirable.

**Comment by gjsjohnmurray** [comment]
There is a precedent for the suggested `"${workspaceFolder:rootA}..."` notation. It is valid in debug configurations. See [doc here](https://code.visualstudio.com/docs/editor/variables-reference#_variables-scoped-per-workspace-folder) and this code pointer:

https://github.com/microsoft/vscode/blob/cf4922392cb2d657b3be7542626d1f5d54cce286/src/vs/workbench/services/configurationResolver/common/variableResolver.ts#L182

**Comment by andreamah** [comment]
> Hmmm, there have been several requests for this feature but they all keep timing out due to lack of votes. Is there a way to unify all these and get enough votes to get this in?

This is already considered as a feature request and is in our backlog. Issues on the "Backlog Candidates" milestone are ones that are evaluated based on number of upvotes.

**Comment by Dtreschl** [comment]
If I am understanding the issue here properly, although not a complete fix, I've been able to exclude a root folder locally via a slightly hacky solution.

Adding the root folder in question (let's call it folder_1) to any of the VSCode search settings doesn't seem to remove it from the search, as noted above. And folder_1 contains many thousands of files, so it doubles my search load time. It's mostly got .c, .h, & .py files in it. To exclude those from my results, I put a .vscode folder into folder_1, and a settings.json inside of that with the following contents:
`<root_dir/folder_1/.vscode/settings.json>`
`{
   "search.exclude": {
      "**/*.c": true,
      "**/*.h": true,
      "**/*.py": true
   }
}`

My "files to include line" in my search editor is `*.c, *.h, *.py` , so by excluding those types from folder_1, they don't get included in my search results. Search time is halved, and it won't search for those file types within folder_1 any more. I admit, it's a hacky solution, as a repository with many file types would make this slightly clunky. But if anyone is looking for a temporary fix to this, hopefully I've been of help. It does not seem that you are able to just exclude `**/*.*`, at least on my end.

**Comment by starball5** [comment]
Somewhat related: #82145.

Related on VS Code: [How can I exclude files from one of a multi-root workspace's root folders from search in VS Code?](https://stackoverflow.com/q/76801657/11107541)

**Comment by jcrben** [comment]
Does this also include files? I'm not having luck excluding a root file - `package-lock.json`.  
![image](https://github.com/microsoft/vscode/assets/5614134/438f8cc9-20e2-4f9a-93fe-787dbe69a61e)
